numpy/_core/src/common/get_attr_string.h

```diff
@@ -46,22 +47,19 @@ _is_basic_python_type(PyTypeObject *tp)
  *
  * In future, could be made more like _Py_LookupSpecial
```

Comment:
seems like you're moving the API away from `_PyArray_LookupSpecial`, which returns a `PyObject *`. Maybe delete this comment if it no longer applies?

Reply:
Good point, actually `_PyObject_LookupSpecial` *is* the old name for `PyObject_GetOptionalAttr`.  But I didn't time if it makes sense to just remove the exact type-check fast path or not.  So going to keep this (with the adaptation for now).

Reply:
(More importantly: You need Python 3.12 to get the super-fast behavior for `type(...)`, so we actually can't fully replace it yet.)